### PR TITLE
Handle Horizon response

### PR DESCRIPTION
**Pull request by juanhurtado10**
This is part of #23

### CODE REVIEW

File: test/horizon/client/default_test.exs [hunk 1]
```diff
@@ -10,42 +12,58 @@ defmodule Stellar.Horizon.Client.CannedHTTPClient do
           body :: String.t(),
           options :: list()
         ) :: {:ok, map()} | {:error, map()}
```
Comment (luishurtado) [inline test/horizon/client/default_test.exs:14]: We should include all possible status codes that Horizon could return. For example, the 50x status.

**Review by luishurtado — CHANGES_REQUESTED** [review]

File: lib/horizon/client/default.ex [hunk 2]
```diff
@@ -1,22 +1,56 @@
...
+    |> handle_response()
+  end
+
+  @spec handle_response(response :: client_response()) :: parsed_response()
+  defp handle_response({:ok, status, _headers, body}) when status >= 200 and status <= 299 do
+    decoded_body = json_library().decode!(body, keys: :atoms)
+    {:ok, decoded_body}
+  end
+
+  defp handle_response({:ok, status, _headers, body}) when status >= 300 and status <= 599 do
```
[the remark this answers is not in the document]
Comment (juanhurtado10) [reply lib/horizon/client/default.ex]: You are right, redirects are configured to be followed, they should not be treated as errors. Changes were done!

File: test/horizon/client/default_test.exs (hunk 1 above)
Comment (juanhurtado10) [reply test/horizon/client/default_test.exs:14]: All possible statuses were added.

File: test/support/fixtures/horizon/400.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  "status": 400,
+  "detail": "The transaction failed when submitted to the stellar network. The `extras.result_codes` field on this response contains further details.  Descriptions of each code can be found at: https://www.stellar.org/developers/guides/concepts/list-of-operations.html",
+  "extras": {
+    "envelope_xdr": "AAAAAgAAAADdfhHDs4Vaug6p8Oxb1QRjNRdJt3pYKKBVhFHrEgd9QAAAAAoAEi4YAAAAAwAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAACwAAAAAAAAAAAAAAAAAAAAESB31AAAAAQFhc/liVXbLk3NtB2BtweFJ064JdDIfrTSrqKMhb1oIRK+0PSyvjzZTkRCJmQY3bHNXYNuepa2TF7aBdibrb1gI=",
+    "result_codes": {
+      "transaction": "tx_insufficient_fee"
+    },
+    "result_xdr": "AAAAAAAAAAr////3AAAAAA=="
+  }
+}
```
Comment (luishurtado) [inline test/support/fixtures/horizon/400.json]: No newline at the end of the file

File: test/horizon/client/default_test.exs [hunk 4]
```diff
@@ -12,21 +12,42 @@ defmodule Stellar.Horizon.Client.CannedHTTPClient do
...
 
   def request(:get, @base_url <> "/transactions/" <> _hash, _headers, _body, _opts) do
-    json_body = Horizon.fixture(:transaction)
+    json_body = Horizon.fixture("200")
     {:ok, 200, [], json_body}
   end
 
   def request(:post, @base_url <> "/transactions?tx=bad", _headers, _body, _opts) do
-    json_error = Horizon.fixture(:error)
+    json_error =  Horizon.fixture("400")
```
Comment (luishurtado) [inline test/horizon/client/default_test.exs]: Extra space after the `=`

**Review by luishurtado — CHANGES_REQUESTED** [review]

File: test/horizon/client/default_test.exs (hunk 4 above)
Comment (juanhurtado10) [reply test/horizon/client/default_test.exs]: This was solved yesterday, I think you're reviewing an old version of the PR!

**Review by luishurtado — APPROVED** [review]

**Pull request merged**